### PR TITLE
Fix translation issues

### PR DESCRIPTION
**Pull request by fiwa**
Closes https://github.com/Aventyret/solarplexus/issues/60

## Problem
When running `wp i18n make-json` separate files are created based on where the strings are extracted. The naming convention for each source file is {textdomain}-{locale}-{md5(source)}.json.

How WordPress looks for translation files:

1. _{textdomain}-{locale}-{script-handle}.json_
2. _{textdomain}-{locale}-{source}.json_
3. _{textdomain}-{locale}-{md5(source)}.json_

"Source" in this case is _build/index.js_ added with the handle `solarplexus-script`. This results in translations extracted from others sources not being loaded. 

**Example**
Translations extracted from _src/components/dynamic-inspector-controls.js_ ends up in a file named _{textdomain}-{locale}-{md5(src/components/dynamic-inspector-controls.js)}.json_. The file is not loaded separately, resulting in the translations for the file not being loaded.

## Solution
Translations needs to be extracted from "build/index.js" BUT that does not work on the minified version. Instead the translation scripts needs to be run in the `yarn start` state.

**Steps to update translations**

1. Run `yarn start` to create a non-minified build/index.js.
2. Run `yarn translate` to update splx.pot and existing splx-{locale}.po files.
3. Translate new strings in splx-{locale}.po. This can be done through [Poedit](https://poedit.net) or directly in the file.
4. Run `yarn translate:compile` to update splx-{locale}.mo and json files.

An alternative to the steps above could be to use [po2json](https://github.com/mikeedwards/po2json) to convert the mo files to json. But currently the package generates [warnings](https://github.com/mikeedwards/po2json/issues/103) with Node 16.

### CODE REVIEW

**Review by perarnborg — COMMENTED** [review]
Smart thinking!

`wp i18n update-po` fails for me. Perhaps wp-cli need to be added to the project as a dependency to ensure correct version? Just a guess.... It works for you @fiwa ?

File: package.json [hunk 1]
```diff
@@ -24,7 +24,13 @@
 		"start": "wp-scripts start",
 		"docs:dev": "vitepress dev docs",
 		"docs:build": "vitepress build docs",
-		"docs:serve": "vitepress serve docs"
+		"docs:serve": "vitepress serve docs",
+		"translate": "npm run translate:pot && npm run translate:update",
```
[the remark this answers is not in the document]
Comment (fiwa) [reply package.json:28]: Yes of course, I will ta a look at that!

[the remark this answers is not in the document]
Comment (fiwa) [reply package.json]: @perarnborg Suggested changes are in place. Are you able to run the `wp-cli` commands after `composer install`?

**Comment by perarnborg** [comment]
@fiwa Perfect, it works!

Do you think it would be better to use po2json? Because we COULD bump down node to v14. If not I think this should be merged!

**Comment by fiwa** [comment]
> @fiwa Perfect, it works!
> 
> Do you think it would be better to use po2json? Because we COULD bump down node to v14. If not I think this should be merged!

@perarnborg  I think an extra step in the translation process is worth it in order to use Node 16.

**Pull request merged**